### PR TITLE
Update Go to 1.21 and deps for k8s 1.27

### PR DESCRIPTION
**Pull request by pmatseykanets**
Ref: https://github.com/rancher/kube-api-auth/pull/20#pullrequestreview-1844162604

Original ticket https://github.com/rancher/rancher/issues/43318

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,250 +1,257 @@
...
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.27.10
+	oras.land/oras-go => oras.land/oras-go v1.2.2 // for docker 20.10.x compatibility
 )
 
 require (
 	github.com/gorilla/mux v1.8.0
-	github.com/rancher/norman v0.0.0-20230426211126-d3552b018687
-	github.com/rancher/rancher v0.0.0-20230712102934-01a8529371b2
+	github.com/rancher/norman v0.0.0-20240205154641-a6a6cf569608
+	github.com/rancher/rancher v0.0.0-20240205190724-2f7113dc32d4
```
Comment (pmatseykanets) [inline go.mod:66]: Points to https://github.com/rancher/rancher/releases/tag/v2.8.2

File: go.mod [hunk 2]
```diff
@@ -1,250 +1,257 @@
...
+	k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.27.10
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.27.10
+	oras.land/oras-go => oras.land/oras-go v1.2.2 // for docker 20.10.x compatibility
 )
 
 require (
 	github.com/gorilla/mux v1.8.0
-	github.com/rancher/norman v0.0.0-20230426211126-d3552b018687
-	github.com/rancher/rancher v0.0.0-20230712102934-01a8529371b2
+	github.com/rancher/norman v0.0.0-20240205154641-a6a6cf569608
```
Comment (pmatseykanets) [inline go.mod:65]: Points to https://github.com/rancher/norman/commit/a6a6cf5696088c32002953d36b75bdcc84f2399e

**Review by pjbgf — APPROVED** [review]
LGTM

**Review by tomleb — APPROVED** [review]

**Review by macedogm — APPROVED** [review]

**Review by macedogm — APPROVED** [review]

**Review by pjbgf — APPROVED** [review]

**Review by crobby — APPROVED** [review]
Version bumps all seem reasonable.

**Pull request merged**